src/agenda/reservation-list/index.tsx

```diff
@@ -107,38 +107,47 @@ class ReservationList extends Component<ReservationListProps, State> {
   }
 
   componentDidUpdate(prevProps: ReservationListProps) {
-    if (this.props.topDay && prevProps.topDay && prevProps !== this.props) {
+    if (this.props.topDay && prevProps.topDay) {
       if (!sameDate(prevProps.topDay, this.props.topDay)) {
         this.setState({reservations: []},
-          () => this.updateReservations(this.props)
+          () => this.updateReservations(this.props, prevProps) // Pass prevProps
         );
-      } else {
-        this.updateReservations(this.props);
+      } else if (
+        this.props.items !== prevProps.items ||
+        !sameDate(this.props.selectedDay, prevProps.selectedDay)
+      ) {
+        this.updateReservations(this.props, prevProps); // Pass prevProps
       }
     }
   }
-
   updateDataSource(reservations: DayAgenda[]) {
     this.setState({reservations});
   }
 
-  updateReservations(props: ReservationListProps) {
+  updateReservations(props: ReservationListProps, prevProps: ReservationListProps) {
```

Comment:
Why do you need to pass the `prevProps` here if it is not used at `updateReservations`?